sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -183,12 +183,14 @@ Scheduler::GraphBuilder::getOrInsertMemObjRecord(const QueueImplPtr &Queue,
 
   const size_t LeafLimit = 8;
   LeavesCollection::AllocateDependencyF AllocateDependency =
-      [this](Command *Dependant, Command *Dependency, MemObjRecord *Record) {
+      [this](Command *Dependant, Command *Dependency, MemObjRecord *Record,
+             LeavesCollection::EnqueueListT *ToEnqueue) {
         // Add the old leaf as a dependency for the new one by duplicating one
         // of the requirements for the current record
         DepDesc Dep = findDepForRecord(Dependant, Record);
         Dep.MDepCommand = Dependency;
-        Dependant->addDep(Dep);
+        if (Command *ConnectionCmd = Dependant->addDep(Dep))
+          ToEnqueue->push_back(ConnectionCmd);
```

Comment:
My concern here is that 'ToEnqueue' is never checked for being non-nullptr. If it is never nullptr, can the parameter type be changed from 'LeavesCollection::EnqueueListT *' to 'LeavesCollection::EnqueueListT &' ?

Reply:
Fixed.